db/src/tx.rs

```diff
@@ -616,6 +624,17 @@ impl<'conn, 'a> Tx<'conn, 'a> {
                         tx_might_update_metadata = true;
                     }
 
+                    if e == KnownEntid(self.tx_id) && a == entids::DB_TX_INSTANT {
+                        tx_instant_set = true;
+                        if let TypedValue::Instant(instant) = v {
+                            self.tx_instant = instant;
+                        } else {
+                            panic!("This type error should have been caught earlier.");
+                        }
+                        // TODO: INSTANT to be strictly after the last transaction
+                        // timestamp and strictly before the current transactor timestamp.
```

Comment:
We might want to weaken this condition — Datomic can assume correct clocks, but embedded clients cannot.

And we will need a way to bypass this, too, for syncing.